cborg/src/Codec/CBOR/Read.hs

```diff
@@ -1395,7 +1395,7 @@ isInt64Canonical sz i#
 
 {-# INLINE tryConsumeWord #-}
 tryConsumeWord :: Word8 -> ByteString -> DecodedToken Word
-tryConsumeWord hdr !bs = case word8ToWord hdr of
+tryConsumeWord hdr !bs = case hdr of
```

Comment:
No no no. This is actually essential. Drop this whole patch. Sorry.

These conversion to `Word` are essential because GHC handles `case` on `Int` and `Word` using jump tables, but all the other `Num` types like `Int8`, `Word8` etc it uses massive chains of `(==)` tests. So the conversion to `Word` is essential for good performance here, O(1) vs O(n).

Reply:
See also https://github.com/well-typed/cborg/blob/master/cborg/src/Codec/CBOR/Read.hs#L1429